src/main/java/org/galatea/starter/service/PriceService.java

```diff
@@ -20,13 +42,101 @@
+  public Collection<StockPrices> filteredPrices;
+
+  public Collection<StockPrices> getPricesByStock(String stock, String daysToLookBack) {
+
+    processStartTime = System.currentTimeMillis();
+    String size;
+
+    //Determine the response size from Alpha Vantage based on daysToLookBack
+    Integer days = Integer.parseInt(daysToLookBack);
+    log.info("\n number of days to look back: " + daysToLookBack);
+    if (days < 101) {
+      size = "compact";
+    } else {
+      size = "full";
+    }
+
+    //Call Alpha Vantage API based on (stock, size)
+    AlphaPrices objPrices = pricesclient.getPricesByStock(stock, size);
+
+    //Convert to internal Price Objects
+    ArrayList<StockPrices> convertedPrices = ConvertPrices(stock, objPrices);
+
+    //Filter response size, starting with T=1
+    Collections.sort(convertedPrices, Comparator.comparing(StockPrices::getDate).reversed());
+    filteredPrices = convertedPrices.subList(1, days);
+
+    log.info("Response from Alpha Vantage for stock: {} and the response size: {}. Processing Time was {}, response object: {}.", stock, size, processTime, filteredPrices);
+    return filteredPrices;
   }
-}
-  //place holder for switch based on # of days requested
 
 
+  public ArrayList<StockPrices> ConvertPrices(String stock, AlphaPrices objPrices) {
+
+    StockPrices dataPoints;
+    ArrayList<StockPrices> converted = new ArrayList<>();
+
+
+    for (Entry<Date, ResponsePrices> entry : objPrices.getAvPrices()
+        .entrySet()) {
 
+      Date dates = entry.getKey();
+      ResponsePrices internalPrices = entry.getValue();
+      StockPricesBuilder builder = StockPrices.builder();
+      builder.date(dates);
+      builder.adjustedClose(internalPrices.getAdjustedClose());
+      builder.close(internalPrices.getClose());
+      builder.high(internalPrices.getHigh());
+      builder.close(internalPrices.getClose());
+      builder.volume(internalPrices.getVolume());
+      builder.splitCoefficient(internalPrices.getSplitCoefficient());
+      builder.low(internalPrices.getLow());
+      builder.stock(stock);
+      dataPoints = builder.build();
+
+      converted.add(dataPoints);
+    }
+    return converted;
+  }
+
+  public String BuildMeta(String stock, String days)
+      throws UnknownHostException, JsonProcessingException {
+
+    StockMetadata stockMetadata;
+    ArrayList<StockMetadata> metadata = new ArrayList<>();
+
+    Long processEndTime = System.currentTimeMillis();
+    processTime = processEndTime - processStartTime;
+
+    Long process = processTime;
+    String processTime = process.toString();
+
+    Long startTime = processStartTime;
+    String start = DateFormat.getInstance().format(startTime);
+
+
+    //Construct Meta Data object
+    StockMetadataBuilder builder = StockMetadata.builder();
+    builder.endpoint("price?stock=" + stock + "&days=" + days);
+    builder.host(InetAddress.getLocalHost().getHostName());
+    builder.responseTime(processTime + "(ms)");
```

Comment:
pass the response time in as a param. you can time the contents of the priceEndpoint method and throw that in.